### PR TITLE
Added pre validation of prometheus URL

### LINKED ISSUE #947: Console issues when Prometheus endpoint is configured but Prometheus cannot be reached
minio --version
minio version RELEASE.2021-08-05T22-01-19Z

Steps to repro:
1. Have prometheus endpoint configured for MinIO console (e.g. `MINIO_PROMETHEUS_URL=http://localhost:9090`)
2. Start minio when prometheus is not available (not started, network disconnect etc)
3. Log in to console
4. Dashboard shows all spinning icons (as expected)
5. Navigate to license page (or pages listed below)
  - Notice that the page never loads.  If you try to refresh the page, it hangs.  Pressing stop button in the browser causes full log out of console

If you start/reconnect to prometheus and refresh, things work as expected.

Observed it affecting the following pages:

Object browser
Account
Buckets
Users
Groups
IAM policies
License

### PR DESCRIPTION
**Pull request by bexsoft**
fixes #947 

## What does this do?

Adds a pre-validation to prometheus URL, so we can avoid long timeouts & unnecessary requests that will fail if URL is unreachable

## How does it look?
Screen Shot 2021-08-16 at 23 45 18

### CODE REVIEW

File: restapi/admin_info.go [hunk 1]
```diff
@@ -871,10 +873,39 @@ func unmarshalPrometheus(endpoint string, data interface{}) bool {
...
+	_, err := net.DialTimeout("tcp", urlQuery, timeout)
+
+	if err != nil {
+		LogError("Non reachable Prometheus URL: %s (%v)", urlQuery, err)
+		return false
+	}
+
+	return true
+}
+
```
Comment (harshavardhana) [inline restapi/admin_info.go:898]: Do not attempt a non-TLS dial @bexsoft we should do a full HTTP call such as make a call to 

```
ctx, cancel := context.WithTimeout(context.Background(), 2 * time.Second)
defer cancel()
...
req, err := http.NewRequestWithTimeout(ctx, http.MethodGet, URL+"/-/healthy", nil)
...

then use 
GetConsoleHTTPClient().Do(req)
```

**Review by dvaldivia — APPROVED** [review]

File: restapi/admin_info.go (hunk 1 above)
Comment (bexsoft) [reply restapi/admin_info.go]: Sounds good, changes applied

**Review by dvaldivia — APPROVED** [review]

File: restapi/admin_info.go [hunk 2]
```diff
@@ -871,10 +873,42 @@ func unmarshalPrometheus(endpoint string, data interface{}) bool {
...
+		LogError("Non reachable Prometheus URL: %s (%v)", url, err)
+		return false
+
+	}
+
+	if response.StatusCode == http.StatusOK {
+		return true
+	}
+
+	return false
```
Comment (harshavardhana) [inline restapi/admin_info.go:899]:
```suggestion
        return response.StatusCode == http.StatusOK
```

File: restapi/admin_info.go [hunk 3]
```diff
@@ -871,10 +873,42 @@ func unmarshalPrometheus(endpoint string, data interface{}) bool {
...
+	defer cancel()
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url+"/-/healthy", nil)
+
+	if err != nil {
+		LogError("Error Building Request: (%v)", err)
+		return false
+	}
+
+	response, err2 := GetConsoleHTTPClient().Do(req)
```
Comment (harshavardhana) [inline restapi/admin_info.go:887]:
```suggestion
	response, err := GetConsoleHTTPClient().Do(req)
```
Simplify

File: restapi/admin_info.go [hunk 4]
```diff
@@ -871,10 +873,42 @@ func unmarshalPrometheus(endpoint string, data interface{}) bool {
...
+
+	if err2 != nil {
```
Comment (harshavardhana) [inline restapi/admin_info.go:889]:
```suggestion
	if err != nil {
```

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: restapi/admin_info.go (hunk 2 above)
Comment (bexsoft) [reply restapi/admin_info.go]: done

File: restapi/admin_info.go (hunk 4 above)
Comment (bexsoft) [reply restapi/admin_info.go]: done

File: restapi/admin_info.go (hunk 3 above)
Comment (bexsoft) [reply restapi/admin_info.go]: done

**Review by dvaldivia — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Comment by wasilak** [comment]
Hey guys, new validation broke integration with [VictoriaMetrics](https://github.com/VictoriaMetrics/VictoriaMetrics), which is Prometheus compatible database, but unfortunately doesn't have `/-/healthy` endpoint.

**Pull request merged**